framework/src/geomsearch/FindContactPoint.C

```diff
@@ -54,70 +55,49 @@ findContactPoint(PenetrationInfo & p_info,
 
   const Elem * side = p_info._side;
 
-  const std::vector<Point> & phys_point = _fe->get_xyz();
+  const std::vector<Point> & phys_point = fe_side->get_xyz();
 
-  const std::vector<RealGradient> & dxyz_dxi = _fe->get_dxyzdxi();
-  const std::vector<RealGradient> & d2xyz_dxi2 = _fe->get_d2xyzdxi2();
-  const std::vector<RealGradient> & d2xyz_dxieta = _fe->get_d2xyzdxideta();
+  const std::vector<RealGradient> & dxyz_dxi = fe_side->get_dxyzdxi();
+  const std::vector<RealGradient> & d2xyz_dxi2 = fe_side->get_d2xyzdxi2();
+  const std::vector<RealGradient> & d2xyz_dxieta = fe_side->get_d2xyzdxideta();
 
-  const std::vector<RealGradient> & dxyz_deta = _fe->get_dxyzdeta();
-  const std::vector<RealGradient> & d2xyz_deta2 = _fe->get_d2xyzdeta2();
-  const std::vector<RealGradient> & d2xyz_detaxi = _fe->get_d2xyzdxideta();
+  const std::vector<RealGradient> & dxyz_deta = fe_side->get_dxyzdeta();
+  const std::vector<RealGradient> & d2xyz_deta2 = fe_side->get_d2xyzdeta2();
+  const std::vector<RealGradient> & d2xyz_detaxi = fe_side->get_d2xyzdxideta();
 
   if (dim == 1)
   {
-    const Node * left = master_elem->node_ptr(0);
-    const Node * right = left;
-    Real leftCoor = (*left)(0);
-    Real rightCoor = leftCoor;
-    for (unsigned i = 1; i < master_elem->n_nodes(); ++i)
-    {
-      const Node * curr = master_elem->node_ptr(i);
-      Real coor = (*curr)(0);
-      if (coor < leftCoor)
-      {
-        left = curr;
-        leftCoor = coor;
-      }
-      if (coor > rightCoor)
-      {
-        right = curr;
-        rightCoor = coor;
-      }
-    }
-    const Node * nearestNode = left;
-    Point nearestPoint(leftCoor, 0, 0);
-    if (side->node(0) == right->id())
-    {
-      nearestNode = right;
-      nearestPoint(0) = rightCoor;
-    }
-    else if (side->node(0) != left->id())
-    {
-      mooseError("Error findContactPoint.  Logic error in 1D");
-    }
-    p_info._closest_point_ref = FEInterface::inverse_map(dim, _fe_type, master_elem, nearestPoint, TOLERANCE, false);
-    p_info._closest_point = nearestPoint;
-    p_info._normal = Point(left == nearestNode ? -1 : 1, 0, 0);
+    const Node * nearest_node = side->node_ptr(0);
+    p_info._closest_point = *nearest_node;
+    p_info._closest_point_ref = master_elem->master_point(master_elem->get_node_index(nearest_node));
+    std::vector<Point> elem_points = {p_info._closest_point_ref};
+    fe_elem->reinit(master_elem, &elem_points);
+
+    const std::vector<RealGradient> & elem_dxyz_dxi = fe_elem->get_dxyzdxi();
+    p_info._normal = elem_dxyz_dxi[0];
+    if (nearest_node->id() == master_elem->node(0))
```

Comment:
`Elem::node()` is deprecated, please use the better-named `Elem::node_id()` instead.